src/magic.cpp

```diff
@@ -906,6 +918,22 @@ bool spell::is_valid_effect_target( valid_target t ) const
     return type->effect_targets[t];
 }
 
+bool spell::target_by_monster_id( Creature &target ) const
+{
+    bool valid = false;
+    monster *const mon = dynamic_cast<monster *>( &target );
```

Comment:
dynamic cast bad. use `as_monster()` instead

Reply:
Theres no creature as_monster(). So I just changed the function to take the targets tripoint as an argument. Hope thats Ok.